schema.cc

```diff
@@ -840,6 +855,11 @@ bool thrift_schema::is_dynamic() const {
     return _is_dynamic;
 }
 
+schema_builder& schema_builder::with_partitioner(sstring name, unsigned shard_count, unsigned sharding_ignore_msb_bits) {
+    _raw._partitioner = get_partitioner(name, shard_count, sharding_ignore_msb_bits);
+    return *this;
```

Comment:
What does it mean to associate a schema with shard_count and sharding_ignore_msb_bits? These are per-node settings. They aren't shared between different nodes for the same table.

Reply:
Before we hashed the partitioner into the schema, it wasn't a problem, but now it is. Unless we only hash the partitioner name.

Reply:
We only hash partitioner name. I agree that shard_count and sharding_ignore_msb_bits are per node. In this PR it's in schema because partitioner contains it. I'm preparing another PR on top of this one to separate partitioners and sharding logic for good.

Reply:
Then everything's fine.